src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extensions.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo } from 'react';
+import {
+  QueryEditorExtension,
+  QueryEditorExtensionConfig,
+  QueryEditorExtensionDependencies,
+} from './query_editor_extension';
+
+interface QueryEditorExtensionsProps extends QueryEditorExtensionDependencies {
+  configMap?: Record<string, QueryEditorExtensionConfig>;
+  componentContainer: Element;
+  bannerContainer: Element;
+}
+
+const QueryEditorExtensions: React.FC<QueryEditorExtensionsProps> = React.memo((props) => {
```

Comment:
nit: is there a better name for this given its both a collection of extensions as well as specifying their destinations (and its easy to confuse components that are one character apart). perhaps its a manager as its responsibilities could include managing what gets rendered if we want more rules than rendering everything

Reply:
maybe `QueryEditorExtensionManager` and `QueryEditorExtension`? but I'm not very sure if a manager is something that should be rendered on the page, current impl needs `QueryEditor` to render `QueryEditorExtensions`. The one character naming existed in other places, like `suggestion_component` and `suggestions_component`, although i do find it to be confusing